docs/h5py_dataset.rst

```diff
@@ -105,11 +111,73 @@ with training examples and the last 10 rows with test examples.
 >>> targets[...] = numpy.vstack([train_targets, test_targets])
 
 The last thing we need to do is to give :class:`~.datasets.hdf5.H5PYDataset`
-a way to recover what the splits are. This is done by setting attributes in
-the root group.
+a way to recover what the splits are. This is done by setting the ``split``
+attribute of the root group.
+
+>>> split_array = numpy.empty(
+...     6,
+...     dtype=numpy.dtype([
+...         ('split', 'a', 5),
+...         ('source', 'a', 15),
+...         ('start', numpy.int64, 1), ('stop', numpy.int64, 1),
+...         ('available', numpy.bool, 1),
+...         ('comment', 'a', 1)]))
+>>> split_array[0:3]['split'] = 'train'.encode('utf8')
+>>> split_array[3:6]['split'] = 'test'.encode('utf8')
+>>> split_array[0:6:3]['source'] = 'vector_features'.encode('utf8')
+>>> split_array[1:6:3]['source'] = 'image_features'.encode('utf8')
+>>> split_array[2:6:3]['source'] = 'targets'.encode('utf8')
+>>> split_array[0:3]['start'] = 0
+>>> split_array[0:3]['stop'] = 90
+>>> split_array[3:6]['start'] = 90
+>>> split_array[3:6]['stop'] = 100
+>>> split_array[:]['available'] = True
+>>> split_array[:]['comment'] = '.'.encode('utf8')
+>>> f.attrs['split'] = split_array
+
+We created a 1D numpy array with six elements. The ``dtype`` for this array
+is a compound type: every element of the array is a tuple of ``(str, str, int,
```

Comment:
Maybe mention that the strs are bytes on Python 3? Or do you handle this somewhere else?